src/java_bytecode/generate_java_generic_type.cpp

```diff
@@ -43,8 +44,38 @@ symbolt generate_java_generic_typet::operator()(
   const java_class_typet &class_definition =
     to_java_class_type(pointer_subtype);
 
-  const irep_idt generic_name =
-    build_generic_name(existing_generic_type, class_definition);
+  // Small auxiliary function, to print a single argument name.
+  std::function<std::string(reference_typet)> build_argument_name =
+    [&](const reference_typet &argument)
+    {
+      const irep_idt &id(id2string(argument.subtype().get(ID_identifier)));
+      if(is_java_array_tag(id))
+      {
+        std::string name;
+        name.append(pretty_print_java_type(id2string(id)));
```

Comment:
why create an empty string and append to it? Better create the initial string with that content.